Modules/Sources/WordPressCore/DataStore/InMemoryDataStore.swift

```diff
@@ -2,20 +2,42 @@ import Foundation
 @preconcurrency import Combine
 
 /// A `DataStore` type that stores data in memory.
-public protocol InMemoryDataStore: DataStore {
+public actor InMemoryDataStore<T: Sendable & Identifiable>: DataStore {
+
+    public struct Query: Sendable {
+        // TODO: Replace this with `Predicate` once iOS 17 becomes the minimal deployment target.
+        let filter: @Sendable (T) -> Bool
+        let sortBy: any SortComparator<T>
+
+        public init(sortBy: any SortComparator<T>, filter: @escaping @Sendable (T) -> Bool) {
+            self.sortBy = sortBy
+            self.filter = filter
+        }
+    }
+
     /// A `Dictionary` to store the data in memory.
-    var storage: [T.ID: T] { get set }
+    private var storage: [T.ID: T] = [:]
```

Comment:
How does it work with ordered collections (order based on the order returned from the server)?

Reply:
I think you can put the order into the model type. That's probably how you would do it if the model is stored in Core Data or a SQLite database?

For example, in WordPress.org plugin directory, there are a few categories like featured plugins and recommended plugins. You probably would design the model like this:

```
struct PluginDirectoryCategry {
  var plugin: PluginInformation
  var kind: Kind /* popular|recommended|... */
  var position: Int
}

let dataStore = InMemeoryDataStore<PluginDirectoryCategry>
```

Reply:
If the plan is to use JSON for storage, JSON has first-class support for arrays, so there is no need to model position using properties and CPU waste cycles on sorting.

Reply:
Does the plugin screen need an `InMemoryDataStore` or any type of store? I'd probably expect it to just load data from the network with an optional HTTP cache and render JSON on screen.

Reply:
> Does the plugin screen need an InMemoryDataStore or any type of store?

When users enter the plugin screen (or any screen), it's always nicer to show them the last-seen content, than a loading indicator. Since plugins rarely change, the cached content would be pretty accurate. So yes, I think we'd want some type of store here.